HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -232,9 +250,20 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
       }
     }
 
-    float HoE = 999.;
-    if (candidate.superCluster()->energy() != 0.)
+    float HoE = 999., sinin = 999.;
+    if (candidate.superCluster()->energy() != 0.) {
       HoE = (*HoverEMap)[candidateRef] / candidate.superCluster()->energy();
+      sinin = (*SigmaIEtaIEtaMap)[candidateRef];
+    }
+
+    if (HoE > egammaHoverECut)
+      continue;
+
+    auto bIt = std::lower_bound(absEtaLowEdges.begin(), absEtaLowEdges.end(), fabs(candidate.eta())) - 1;
+    auto etabinpos = std::distance(absEtaLowEdges.begin(), bIt);
+
```

Comment:
What happens if the candidate `eta` is not within any intervals of `absEtaLowEdges`, for example if `absEtaLowEdges` is misconfigured? (example: eta=1.0 , absEtaLowEdges=[2.0, 2.5])

Is `absEtaLowEdges[0]` always supposed to be zero?

Reply:
For all use cases, we expect the absEtaLowEdges[0] to be zero, indeed. However, theoretically, the user is allowed to use whatever low edge they desire. If you think it is best to constrain the first entry to zero, please let me know and I will do so. For now, I have added a condition that gives a correct logical evaluation for the case you mentioned.